tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_dnd_Clipboard.java

```diff
@@ -0,0 +1,322 @@
+import java.rmi.registry.LocateRegistry;
+import java.rmi.registry.Registry;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BooleanSupplier;
+
+import org.eclipse.swt.dnd.Clipboard;
+import org.eclipse.swt.dnd.RTFTransfer;
+import org.eclipse.swt.dnd.TextTransfer;
+import org.eclipse.swt.dnd.Transfer;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Shell;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import clipboard.ClipboardCommands;
+
+/**
+ * Automated Test Suite for class org.eclipse.swt.dnd.Clipboard
+ *
+ * @see org.eclipse.swt.dnd.Clipboard
+ * @see Test_org_eclipse_swt_custom_StyledText StyledText tests as it also does
+ *      some clipboard tests
+ */
+public class Test_org_eclipse_swt_dnd_Clipboard {
+
+	private static final int REPEAT_COUNT = 3;
+	@TempDir
+	static Path tempFolder;
+	static int uniqueId = 1;
+	private Display display;
+	private Shell shell;
+	private Clipboard clipboard;
+	private TextTransfer textTransfer;
+	private RTFTransfer rtfTransfer;
+	private ClipboardCommands remote;
+	private Process remoteClipboardProcess;
+
+	@BeforeEach
+	public void setUp() {
+		display = Display.getCurrent();
+		if (display == null) {
+			display = Display.getDefault();
+		}
+
+		clipboard = new Clipboard(display);
+		textTransfer = TextTransfer.getInstance();
+		rtfTransfer = RTFTransfer.getInstance();
+	}
+
+	/**
+	 * TODO remove all uses of sleep and change them to processEvents with the
+	 * suitable conditional, or entirely remove them
+	 */
+	private void sleep() throws InterruptedException {
+		SwtTestUtil.processEvents(100, null);
+	}
+
+	/**
+	 * Note: Wayland backend does not allow access to system clipboard from
+	 * non-focussed windows. So we have to create/open and focus a window here so
+	 * that clipboard operations work.
+	 */
+	private void openAndFocusShell() throws InterruptedException {
+		shell = new Shell(display);
+		shell.open();
+		shell.setFocus();
+		sleep();
+	}
+
+	/**
+	 * Note: Wayland backend does not allow access to system clipboard from
+	 * non-focussed windows. So we have to open and focus remote here so that
+	 * clipboard operations work.
+	 */
+	private void openAndFocusRemote() throws Exception {
+		startRemoteClipboardCommands();
+		remote.setFocus();
+		remote.waitUntilReady();
+		sleep();
+	}
+
+	@AfterEach
+	public void tearDown() throws Exception {
+		sleep();
+		try {
+			stopRemoteClipboardCommands();
+		} finally {
+			if (clipboard != null) {
+				clipboard.dispose();
+			}
+			if (shell != null) {
+				shell.dispose();
+			}
+			SwtTestUtil.processEvents();
+		}
+	}
+
+	private void startRemoteClipboardCommands() throws Exception {
+		// TODO: Can I get the jar and run it without extracting it?
+		// TODO: Can I get all the files without having to list them?
```

Comment:
It would indeed be relatively simple with a jar....
But I'm pretty sure that some code exists somewhere (not in SWT nor Eclipse, but more generally on the web), that is able to read the bytes of a Class object or .class file and list all the types that are referenced by it, that could then be filtered; eg `findReferencedTypeNames(ClipboardTest.class).stream().filter(name -> name.startsWith(ClipboardTest.class.getPackageName())`

However, it would be more code for a case that shouldn't become so tricky.

Reply:
As mentioned above simply use the jar or output directory should work, why copy everything into a temp location?

Reply:
Similar to https://github.com/eclipse-platform/eclipse.platform.swt/pull/2541#discussion_r2378890628 I think maintaining the short list of class files is easier than writing a heavier weight mechanism, at least for now.

I may want more later if/when the clipboard tester becomes more complicated, especially if adding tests for more complicated types requires a bunch of extra work here.